generate/templates/manual/repository/statistics.cc

```diff
@@ -0,0 +1,1912 @@
+        m_odbObjectsData.commits.info.erase(itCommitInfo);
+      }
+    }
+    // set new unreachable commits
+    m_odbObjectsData.commits.unreachables = std::move(newUnreachables);
+  }
+}
+
+/**
+ * RepoAnalysis::pruneUnreachableTrees
+ * Prune unreachable trees and decrement reachability of their entries.
+  */
+void RepoAnalysis::pruneUnreachableTrees()
+{
+  while (!m_odbObjectsData.trees.unreachables.empty()) {
+    std::unordered_set<std::string> newUnreachables {};
+
+    // erase unreachable trees
+    for (OdbObjectsData::iterUnreachable itTreeUnrch = m_odbObjectsData.trees.unreachables.begin();
+      itTreeUnrch != m_odbObjectsData.trees.unreachables.end(); ++itTreeUnrch)
+    {
+      OdbObjectsData::iterTreeInfo itTreeInfo = m_odbObjectsData.trees.info.find(*itTreeUnrch);
+    
+      if (itTreeInfo != m_odbObjectsData.trees.info.end()) {
+        // decrease reachability of the entry blobs
+        for (auto &blob : itTreeInfo->second.entryBlobs) {
+          OdbObjectsData::iterBlobInfo itEntryBlobInfo = m_odbObjectsData.blobs.info.find(blob);
+          if (itEntryBlobInfo != m_odbObjectsData.blobs.info.end()) {
+            if (--itEntryBlobInfo->second.reachability == OdbObjectsData::kUnreachable) {
+              m_odbObjectsData.blobs.unreachables.emplace(blob);
+            }
+          }
+        }
+        // decrease reachability of the entry trees and add them as newUnreachables
+        for (auto &treeNameLen : itTreeInfo->second.entryTreesNameLen) {
+          OdbObjectsData::iterTreeInfo itEntryTreeInfo =
+            m_odbObjectsData.trees.info.find(treeNameLen.first);
+          if (itEntryTreeInfo != m_odbObjectsData.trees.info.end()) {
+            if (--itEntryTreeInfo->second.reachability == OdbObjectsData::kUnreachable) {
+              newUnreachables.emplace(treeNameLen.first);
+            }
+          }
+        }
+        // erase tree from the tree's container
+        m_odbObjectsData.trees.info.erase(itTreeInfo);
+      }
+    }
+    // set new unreachable trees
+    m_odbObjectsData.trees.unreachables = std::move(newUnreachables);
+  }
+}
+
+/**
+ * RepoAnalysis::pruneUnreachableBlobs
+ * Rremoves unreachable blobs from their container.
+  */
+void RepoAnalysis::pruneUnreachableBlobs()
+{
+  for (OdbObjectsData::iterUnreachable itBlobUnrch = m_odbObjectsData.blobs.unreachables.begin();
+    itBlobUnrch != m_odbObjectsData.blobs.unreachables.end(); ++itBlobUnrch)
+  {
+    m_odbObjectsData.blobs.info.erase(*itBlobUnrch);
+  }
+}
+
+/**
+ * RepoAnalysis::statsCountAndMax
+ * Statistics for repositorySize (count objects) and biggestObjects (get maximum of them).
+ * Also builds the commits graph.
+ * NOTE: better results achieved not leveraging threads.
+ */
+void RepoAnalysis::statsCountAndMax()
+{
+  // commits
+  for (auto &info : m_odbObjectsData.commits.info) {
+    OdbObjectsData::CommitInfo &commitInfo = info.second;
+    size_t size = commitInfo.size;
```

Comment:
I would search `size_t size` and make all non-member non-param instances const.

Reply:
In this case it can be taken out to the method scope and reused for all the loops